### PR TITLE
feat: Checks if a doc has read-only permissions

### PR DESCRIPTION
**Pull request by edas**
It is needed in cozy-notes, to see if we  should allows
write when displaying a shared document.

### CODE REVIEW

File: packages/cozy-client/src/models/file.js [hunk 1]
```diff
@@ -61,3 +63,13 @@ export function ensureFilePath(file, parent) {
   const path = parent.path + '/' + file.name
   return { path: path, ...file }
 }
+
+/**
+ * Get the id of the parent folder
+ *
+ * @param {object} file  - io.cozy.files document
+ * @returns {string|undefined} id of the parent folder, if any
```
Comment (ptbrowne) [inline packages/cozy-client/src/models/file.js]: What does it mean if there is no parent folder ?

File: packages/cozy-client/src/models/permission.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ * @property {string[]} values
+ * @property {string} type - a couch db database like 'io.cozy.files'
+ */
+
+/**
+ * Is this permission read only ?
+ *
+ * @private
+ * @param {PermissionItem} perm - permission node in a io.cozy.permissions document
+ * @param {object} options -
```
Comment (ptbrowne) [inline packages/cozy-client/src/models/permission.js]: Nit: Throughout this file, there are end of line dashes, I think it's best to remove them.

File: packages/cozy-client/src/models/file.js (hunk 1 above)
Comment (edas) [reply packages/cozy-client/src/models/file.js]: It is usualy that youare at the root folder (or that your files database is somehow broken)

File: packages/cozy-client/src/models/permission.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ * Get the list of permissions blocks
+ * @private
+ */
+async function getOwn(client) {
+  const collection = client.collection('io.cozy.permissions')
+  const data = await collection.getOwnPermissions()
+  const permissions = get(data, 'data.attributes.permissions')
+  if (!permissions) throw `Can't get self permissions`
+  return Object.values(permissions)
+}
```
Comment (ptbrowne) [inline packages/cozy-client/src/models/permission.js]: I think it is clearer when method doing asynchronous things use the verb `fetch`.

File: packages/cozy-client/src/models/file.js (hunk 1 above)
Comment (ptbrowne) [reply packages/cozy-client/src/models/file.js]: Mmh I thought we had an id for the root folder, was I mistaken @nono ?

Comment (nono) [reply packages/cozy-client/src/models/file.js]: We have a root (with the id `io.cozy.files.root-dir`), and it has no parent.

Comment (edas) [reply packages/cozy-client/src/models/file.js]: Oh, I was wrong about the root having an undefined dir_id. It is defined to an empty string.

Let me fix this method so it returns `null` when at root.

File: packages/cozy-client/src/models/permission.js (hunk 2 above)
Comment (edas) [reply packages/cozy-client/src/models/permission.js]: I've  put them toremove prettier warnings about not having descriptions for these param (and well, don't add useless descriptions)

**Pull request merged**